app/controllers/components/course/controller_component_host.rb

```diff
@@ -52,6 +53,20 @@ def gamified?
       def can_be_disabled?
         true
       end
+
+      # Returns a model which the current component can use to interface with its persisted
+      # settings.
+      #
+      # @return [Class] The settings interface class
+      # @return [nil] if the class does not exist
+      def settings_class
+        @settings_class ||= "Course::Settings::#{name.demodulize}".safe_constantize
```

Comment:
Think we should document the `settings_class` method here and indicate where to put the settings classes.